engine/src/main/java/org/terasology/logic/players/FirstPersonHeldItemMountPointComponent.java

```diff
@@ -15,18 +15,62 @@
  */
 package org.terasology.logic.players;
 
+import jopenvr.VRControllerState_t;
+import org.joml.Matrix4f;
 import org.terasology.entitySystem.Component;
 import org.terasology.entitySystem.Owns;
 import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.math.TeraMath;
 import org.terasology.math.geom.Vector3f;
+import org.terasology.rendering.openvrprovider.ControllerListener;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.rendering.openvrprovider.OpenVRProvider;
 
 /**
  * Only used by the client side so that held items can be positioned in line with the camera
  */
-public class FirstPersonHeldItemMountPointComponent implements Component {
+public class FirstPersonHeldItemMountPointComponent implements Component, ControllerListener {
+
     @Owns
     public EntityRef mountPointEntity = EntityRef.NULL;
     public Vector3f rotateDegrees = Vector3f.zero();
     public Vector3f translate = Vector3f.zero();
+    public Quat4f rotationQuaternion;
     public float scale = 1f;
+
+    // TODO: @In
+    private final OpenVRProvider vrProvider = OpenVRProvider.getInstance();
+
+    // The hand/tool models seem to have an origin other than the pivot point. This is a best-effort correction,
+    // in the form of a 4x4 homogeneous transformation matrix
+    private Matrix4f toolAdjustmentMatrix = new Matrix4f(
+            1.0f, 0.0f, 0.0f, 0.0f,
+            0.0f, (float) Math.cos(230.* TeraMath.DEG_TO_RAD), (float) Math.sin(230. * TeraMath.DEG_TO_RAD), 0.0f,
+            0.0f, (float) -Math.sin(230. * TeraMath.DEG_TO_RAD), (float) Math.cos(230. * TeraMath.DEG_TO_RAD), 0.0f,
+            0.0f, -0.05f, -0.2f, 1.0f
+    );
+
+    public void trySubscribeToControllerPoses() {
```

Comment:
I think the code can be considered stable enough now. Can you please write some javadocs for the public methods of this class?